tests/test_auto/test_tune.py

```diff
@@ -0,0 +1,210 @@
+        for i, pipeline in enumerate((Pipeline(NaiveModel(j), horizon=7) for j in range(10)))
+    ]
+
+
+def test_objective(
+    example_tsds,
+    target_metric=MAE(),
+    metric_aggregation: Literal["mean"] = "mean",
+    metrics=[MAE()],
+    backtest_params={},
+    initializer=MagicMock(spec=_Initializer),
+    callback=MagicMock(spec=_Callback),
+    relative_params={
+        "_target_": "etna.pipeline.Pipeline",
+        "horizon": 7,
+        "model": {"_target_": "etna.models.NaiveModel", "lag": 1},
+    },
+    pipeline=Pipeline(NaiveModel()),
+):
+    trial = MagicMock(relative_params=relative_params)
+    _objective = Tune.objective(
+        ts=example_tsds,
+        pipeline=pipeline,
+        target_metric=target_metric,
+        metric_aggregation=metric_aggregation,
+        metrics=metrics,
+        backtest_params=backtest_params,
+        initializer=initializer,
+        callback=callback,
+    )
+    aggregated_metric = _objective(trial)
+    assert isinstance(aggregated_metric, float)
+
+    initializer.assert_called_once()
+    callback.assert_called_once()
+
+
+def test_fit(
+    ts=MagicMock(),
+    tune=MagicMock(),
+    timeout=4,
+    n_trials=2,
+    initializer=MagicMock(),
+    callback=MagicMock(),
+):
+    Tune.fit(
+        self=tune,
+        ts=ts,
+        timeout=timeout,
+        n_trials=n_trials,
+        initializer=initializer,
+        callback=callback,
+    )
+
+    tune._optuna.tune.assert_called_with(
+        objective=tune.objective.return_value, runner=tune.runner, n_trials=n_trials, timeout=timeout
+    )
+
+
+def test_simple_tune_run(example_tsds, optuna_storage, pipeline=Pipeline(NaiveModel(1), horizon=7)):
+    tune = Tune(
+        pipeline,
+        MAE(),
+        metric_aggregation="median",
+        horizon=7,
+        storage=optuna_storage,
+    )
+    tune.fit(ts=example_tsds, n_trials=2)
+
+    assert len(tune._optuna.study.trials) == 2
+    assert len(tune.summary()) == 2
+    assert len(tune.top_k()) == 2
+    assert len(tune.top_k(k=1)) == 1
+
+
+def test_summary(
+    trials,
+    tune=MagicMock(),
+):
+    tune._optuna.study.get_trials.return_value = trials
+    df_summary = AutoBase.summary(self=tune)
+    assert len(df_summary) == len(trials)
+    assert list(df_summary["SMAPE_median"].values) == [trial.user_attrs["SMAPE_median"] for trial in trials]
+
+
+@pytest.mark.parametrize("k", [1, 2, 3])
+def test_top_k(
+    trials,
+    k,
+    tune=MagicMock(),
+):
+    tune._optuna.study.get_trials.return_value = trials
+    tune.target_metric.name = "SMAPE"
+    tune.metric_aggregation = "median"
+    tune.target_metric.greater_is_better = False
+
+    df_summary = AutoBase.summary(self=tune)
+    tune.summary = MagicMock(return_value=df_summary)
+    top_k = AutoBase.top_k(tune, k=k)
+    assert len(top_k) == k
+    assert [pipeline.model.lag for pipeline in top_k] == [i for i in range(k)]  # noqa C416
+
+
+def test_can_handle_uniform(example_tsds, optuna_storage):
```

Comment:
Probably tests `test_can_handle_uniform`, ..., `test_can_handle_categorical` can be parametrized using `pytest.mark.parametrize` because they are very similar.